controllers/hetznercluster_controller.go

```diff
@@ -260,6 +263,34 @@ func (r *HetznerClusterReconciler) reconcileNormal(ctx context.Context, clusterS
 	return reconcile.Result{}, nil
 }
 
+func setControlPlaneEndpoint(hetznerCluster *infrav1.HetznerCluster) bool {
+	if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled {
+		if hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 != "<nil>" {
+			defaultHost := hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4
+			defaultPort := int32(hetznerCluster.Spec.ControlPlaneLoadBalancer.Port)
+
+			if hetznerCluster.Spec.ControlPlaneEndpoint == nil {
+				hetznerCluster.Spec.ControlPlaneEndpoint = &clusterv1.APIEndpoint{
+					Host: defaultHost,
+					Port: defaultPort,
+				}
+				return true
```

Comment:
I just realized we return true here and in the else statement, so in every case.. We can remove these two return true and have one common one after the if else statement, i.e.
```
if hetznerCluster.Spec.ControlPlaneEndpoint == nil{
} else {
}
return true
```